### DISCUSSION TITLE
Consumer genesis generation tooling?

### DISCUSSION THREAD

**Discussion by tbruyelle** [cosmos/interchain-security#1290] [Q&A] [answered]
Is there any tooling/cli to help generate the consumer genesis, in particular the `InitialValSet` field ? 

I imagine something that requests the provider chain for the current validator set, and then registers it with the proper consensus state in the consumer `genesis.json` file. Like a mirror solution of the `gentx/collect-gentx` commands that are used for a sovereign chain to generate its genesis. 

But maybe that's not appropriate ?

**Comment by mpoke** [comment, marked as answer]
Not sure I understand the purpose of such tooling. Currently, the initial val set in the consumer genesis is set by the provider while handling the addition proposal, i.e., https://github.com/cosmos/interchain-security/blob/48a21869e9a56cafff53b5c51077a96d1e3c4926/x/ccv/provider/keeper/proposal.go#L322. Then anybody could just query the consumer genesis (i.e., the CCV part of it).

**Reply by tbruyelle** [reply]
All right, that was the part I missed, indeed no such tools are needed. Thanks @mpoke !
